packages/plugin-typed-css-modules/src/index.ts

```diff
@@ -0,0 +1,57 @@
+import path from 'node:path';
+import type { RsbuildPlugin } from '@rsbuild/core';
+import type { CSSLoaderOptions } from '@rsbuild/shared';
+
+const PLUGIN_NAME = 'rsbuild:typed-css-modules';
+
+export const pluginTypedCSSModules = (): RsbuildPlugin => ({
+  name: PLUGIN_NAME,
+
+  setup(api) {
+    api.modifyBundlerChain({
+      order: 'post',
+      handler: async (chain, { isServer, isWebWorker, CHAIN_ID }) => {
+        if (!isServer && !isWebWorker) {
```

Comment:
Only apply to the `web` target, as `service-worker` should be skipped too.

```suggestion
        if (target === 'web') {
```